pydra/engine/task.py

```diff
@@ -275,15 +276,32 @@ def __init__(
             TODO
 
         """
-        if input_spec is None:
-            input_spec = SpecInfo(name="Inputs", fields=[], bases=(ShellSpec,))
-        self.input_spec = input_spec
-        if output_spec is None:
-            output_spec = SpecInfo(name="Output", fields=[], bases=(ShellOutSpec,))
-
-        self.output_spec = output_spec
+        if input_spec is not None:
+            self.input_spec = input_spec
+        elif not hasattr(self, "input_spec"):
+            self.input_spec = SpecInfo(name="Inputs", fields=[], bases=(ShellSpec,))
+        else:
+            # changing class attribute to instance attribute, so it is part of __dict__
+            # (used in TaskBase.__set/get-state__)
+            self.input_spec = self.input_spec
```

Comment:
this is weird. this should automatically happen in python. with the code that was deleted.

```
In [1]: class A:
   ...:     var1 = None
   ...:     def __init__(self, var1=None):
   ...:         self.var1 = var1
   ...: 

In [2]: a = A()

In [3]: a.__dict__
Out[3]: {'var1': None}
```

Reply:
but my comment was about the class attribute, your example set `var1` also in init